### PR TITLE
Add action to security masks to shout at criminals

### PR DESCRIPTION
**Pull request by Partmedia**
## About the PR 

Add an action to security masks that allows the wearer to shout at criminals when worn.

**Screenshots**
secmask

**Changelog**
:cl: notafet
- add: Security gas masks can be used to shout at criminals.

### CODE REVIEW

**Comment by EmoGarbage404** [comment]
I believe this has been discussed as a concept and i personally am not a fan of it.

**Comment by Partmedia** [comment]
Sorry, I must have missed the discussion. The intent of this is to allow officers to speak while chasing down people because it's otherwise unwieldy to type while on a chase. Otherwise, it is very easy to come across a police chase with no context (i.e. red shirt chasing somebody for no apparent reason). If you have the discussion thread, I'd be happy to continue it there.

**Comment by Emisse** [comment]
I like the idea of it I just think I'm against the precedent this could set going forward. Good PR though.

**Comment by ike709** [comment]
It's just the SS13 sec hailer but for some reason the sound effect isn't included.

**Comment by Partmedia** [comment]
I think I'm missing context from prior discussion; what kind of precedent does this change set?

**Comment by Emisse** [comment]
Over reliance on features that remove need to actually type in typing based game. Macros, emotes, etc. I think this is fine in this case the precedent is just of mild concern.

**Comment by Partmedia** [comment]
I agree that we shouldn't rely on features that remove the need to type. I believe this change warrants an exception, because it is quite difficult to type while chasing someone. Without this change, what ends up happening is that there are "silent chases" where a redshirt is chasing someone without saying anything. Is that really better for roleplay?

**Review by Emisse — APPROVED** [review]

File: Content.Server/Clothing/MaskSystem.cs [hunk 1]
```diff
@@ -98,6 +101,9 @@ private void ToggleMaskComponents(EntityUid uid, MaskComponent mask, EntityUid w
             if (isEquip || !TryComp<BreathToolComponent>(uid, out var breathTool))
                 return;
 
+            if (mask.MaskShoutAction != null)
+                mask.MaskShoutAction.Enabled = !mask.IsToggled;
```
Comment (EmoGarbage404) [inline Content.Server/Clothing/MaskSystem.cs:105]: having the mask toggled to shout doesn't make sense?

**Review by EmoGarbage404 — CHANGES_REQUESTED** [review]
Regardless of my feelings about this balance-wise, having this be tied to the security gas mask feels weird. It doesn't feel intuitive and feels like a very odd thing to tie to a mask. I don't really know what more to say, but conceptually this seems kinda iffy to me.

File: Content.Server/Clothing/MaskSystem.cs (hunk 1 above)
Comment (Partmedia) [reply Content.Server/Clothing/MaskSystem.cs:105]: Toggled means down. It confused me too. Have you tested this code by any chance?

Comment (Partmedia) [reply Content.Server/Clothing/MaskSystem.cs:105]: To clarify, you have to have the mask on in order for the shout action to be enabled. This is consistent with the behavior on SS13.

**Comment by Level10Cybermancer** [comment]
This is probably more obvious in SS13 where the mask also plays a sound effect but the idea is that the mask is playing pre-recorded corporate voicelines in-character as well. Basically like a beepsky on your face.

**Comment by mirrorcult** [comment]
this should only be merged if it comes with the SS13 SFX imo. it should imply "the mask is saying some pre-recorded robotic phrase" and not "the mask is making you say it". could even change the entitysay to be on the mask instead of the person wearing it

**Comment by Partmedia** [comment]
Happy to make these changes. @ike709 would you mind advising on where I might find the SFX and if it is available under a compatible license?

**Comment by ike709** [comment]
It picks one of these at random (some are emag only): https://github.com/tgstation/tgstation/tree/master/sound/runtime/complionator

**Comment by ike709** [comment]
You can see which file name goes with which phrase here: https://github.com/tgstation/tgstation/blob/master/code/datums/hailer_phrase.dm

**Comment by Partmedia** [comment]
Overcome by events.

**Pull request closed without merging**